src/main/java/com/google/devtools/build/lib/rules/cpp/CcModule.java

```diff
@@ -386,9 +390,38 @@ public LibraryToLink createLibraryLinkerInput(
     Artifact dynamicLibrary = nullIfNone(dynamicLibraryObject, Artifact.class);
     Artifact interfaceLibrary = nullIfNone(interfaceLibraryObject, Artifact.class);
 
-    Artifact notNullArtifactForIdentifier = null;
     StringBuilder extensionErrorsBuilder = new StringBuilder();
     String extensionErrorMessage = "does not have any of the allowed extensions";
+
+    PathFragment dynamicLibraryPathFragment = null;
+    if (dynamicLibraryPath != null && !dynamicLibraryPath.isEmpty()) {
+      dynamicLibraryPathFragment = PathFragment.create(dynamicLibraryPath);
+      if (dynamicLibraryPathFragment.isEmpty() || dynamicLibraryPathFragment.isAbsolute() || dynamicLibraryPathFragment.containsUplevelReferences()) {
```

Comment:
Do we need to check dynamicLibraryPathFragment.isEmpty(), given that we already checked dynamicLibraryPath.isEmpty above?

Reply:
Good question. `PathFragment.create` performs normalization if necessary. The docs don't seem to guarantee that normalization never returns an empty `PathFragment` given a non-empty `String`. E.g. `./` or `foo/..` could potentially be normalized away. For reference, the docs say

```
 * <p>Strings are normalized with '.' and '..' removed and resolved (if possible), any multiple
 * slashes ('/') removed, and any trailing slash also removed. Windows drive letters are uppercased.
 * The current implementation does not touch the incoming path string unless the string actually
 * needs to be normalized.
```